app/src/main/java/com/infomaniak/mail/ui/bottomSheetDialogs/ScheduleSendBottomSheetDialog.kt

```diff
@@ -0,0 +1,224 @@
+
+        if (newMessageViewModel.currentMailbox.isFree) {
+            customScheduleItem.setOnClickListener {
+                safeNavigate(
+                    resId = R.id.upgradeProductBottomSheetDialog,
+                    currentClassName = ScheduleSendBottomSheetDialog::class.java.name,
+                )
+            }
+        } else {
+            customScheduleItem.setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    mainViewModel.showSelectDateAndTimeForScheduleDialog()
+                } else {
+                    newMessageViewModel.showSelectDateAndTimeForScheduleDialog()
+                }
+            }
+        }
+
+        fun createScheduleItem(schedule: Schedule): ActionItemView = ActionItemView(this.context).apply {
+            setTitle(schedule.scheduleTitleRes)
+            setDescription(mostDetailedDate(context, date = schedule.date, format = FORMAT_DATE_DAY_MONTH))
+            setIconResource(schedule.scheduleIconRes)
+            setClosingOnClickListener {
+                if (navigationArgs.isAlreadyScheduled) {
+                    navigationArgs.draftResource?.let {
+                        trackScheduleSendEvent(schedule.matomoValue)
+                        mainViewModel.rescheduleDraft(draftResource = it, schedule.date)
+                    }
+                } else {
+                    trackScheduleSendEvent(schedule.matomoValue)
+                    newMessageViewModel.setScheduleDate(schedule.date)
+                    newMessageViewModel.triggerSendMessage()
+                }
+            }
+        }
+
+        val timeToDisplay = TimeToDisplay.getTimeToDisplayFromDate()
+        Schedule.entries.filter { schedule -> timeToDisplay in schedule.timeToDisplay }.forEach { schedule ->
+            scheduleItems.addView(createScheduleItem(schedule))
+        }
+
+        val shouldDisplayDivider = lastScheduleItem.isVisible
+        (scheduleItems.children.first() as ActionItemView).setDividerVisibility(shouldDisplayDivider)
+    }
+
+    companion object {
+        const val MIN_SCHEDULE_DELAY_MINUTES = 5
+        const val MAX_SCHEDULE_DELAY_YEARS = 10
+    }
+}
+
+enum class TimeToDisplay {
+    NIGHT,
+    MORNING,
+    AFTERNOON,
+    EVENING,
+    WEEKEND;
+
+    companion object {
+        fun getTimeToDisplayFromDate(): TimeToDisplay {
+            val now = Date()
+            val timeSlot = Date(now.time)
+            return if (now.isWeekend()) WEEKEND else when (now) {
+                in timeSlot.setHour(0).setMinute(0)..timeSlot.setHour(7).setMinute(54) -> NIGHT
+                in timeSlot.setHour(7).setMinute(55)..timeSlot.setHour(13).setMinute(54) -> MORNING
+                in timeSlot.setHour(13).setMinute(55)..timeSlot.setHour(17).setMinute(54) -> AFTERNOON
+                in timeSlot.setHour(17).setMinute(55)..timeSlot.setHour(23).setMinute(54) -> EVENING
+                else -> NIGHT
+            }
+        }
+    }
+}
```

Comment:
This can be done in an easier way but I'll refactor all this with the snooze because we don't have much more time